src/main/daml/Daml/Finance/Instrument/Swap/Asset/Factory.daml

```diff
@@ -0,0 +1,61 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Instrument.Swap.Asset.Factory where
+
+import DA.Set (singleton)
+import Daml.Finance.Instrument.Swap.Asset.Instrument qualified as Asset (Instrument(..))
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as BaseInstrument (GetCid(..), R, createReference)
+import Daml.Finance.Interface.Instrument.Swap.Asset.Factory qualified as AssetSwap (Create(..), Factory, HasImplementation, Remove(..), View(..))
+import Daml.Finance.Interface.Instrument.Swap.Asset.Types (Asset(..))
+import Daml.Finance.Interface.Types.Common (PartiesMap)
+import Daml.Finance.Interface.Util.Disclosure qualified as Disclosure (I, SetObservers(..), View(..), flattenObservers)
+
+-- | Type synonym for `Factory`.
+type F = Factory
+
+instance AssetSwap.HasImplementation Factory
+-- | Factory template for instrument creation.
+template Factory
+  with
+    provider : Party
+      -- ^ The factory's provider.
+    observers : PartiesMap
+      -- ^ The factory's observers.
+  where
+    signatory provider
+    observer Disclosure.flattenObservers observers
+
+    interface instance AssetSwap.Factory for Factory where
+      asDisclosure = toInterface @Disclosure.I this
+      view = AssetSwap.View with provider
+      create' AssetSwap.Create{asset = Asset{instrument; description; referenceAssetId; issuerPaysFix; fixRate; periodicSchedule; holidayCalendarIds; calendarDataProvider; dayCountConvention; currency; lastEventTimestamp}} = do
+        cid <- toInterfaceContractId <$> create Asset.Instrument
+          with
+            depository = instrument.depository
+            issuer = instrument.issuer
+            issuerPaysFix
+            id = instrument.id
+            version = instrument.version
+            description
+            referenceAssetId
+            fixRate
+            periodicSchedule
+            holidayCalendarIds
+            calendarDataProvider
+            dayCountConvention
+            currency
+            lastEventTimestamp
+            observers
```

Comment:
I think these observers should be passed in from the `Create` choice and not take the observers from the factory contract